src/jti_background.js

```diff
@@ -213,6 +247,75 @@ function addTemplate (templateName, issueTypeField, projectsField, text, callbac
     });
 }
 
+function addDomain (domainName, callback) {
+    chrome.storage.sync.get(StorageID, function (data) {
+        var domainJSON = {};
+
+        if (data[StorageID]) {
+            domainJSON = data[StorageID];
+        }
+
+        var newDomain = {
+            'id': getNextID(domainJSON.options.domains),
+            'name': domainName
+        };
+
+        validateDomain(domainName, function (message) {
+            if (message) {
+                callback(false, message);
+            } else {
+                domainJSON.options.domains[newDomain.id] = newDomain;
+                // Refresh existing pages with this URL.
+                matchRegexToJsRegex(domainName);
+                chrome.tabs.query({windowId: chrome.windows.WINDOW_ID_CURRENT}, function (tabs) {
+                    $.each(tabs, function (tabIndex, tab) {
+                        var chromeRegex = new RegExp('chrome://extensions');
+                        if (matchRegexToJsRegex(domainName).test(tab.url) && (!chromeRegex.test(tab.url))) {
+                            chrome.tabs.reload(tab.id);
+                        }
+                    });
+                    saveTemplates(domainJSON, callback, newDomain.id);
+                });
+            }
+        });
+    });
+}
+
+function removeDomain (domainID, removeAll, callback) {
+    chrome.storage.sync.get(StorageID, function (data) {
+        if (data[StorageID]) {
+            var domainJSON = data[StorageID];
+            $.each(data[StorageID].options.domains, function (index, domain) {
```

Comment:
I think it would be simpler to have an `if/else` here instead of a loop.
```
if (removeAll) {
    domainJSON.options.domains = [];
} else {
    delete domainJSON.options.domains[domainID];
}
```

Reply:
:+1: good call, we can reference the ID now that we're not deleting based on name.